astro/src/content/quickstarts/quickstart-ruby-on-rails-api.mdx

```diff
@@ -0,0 +1,351 @@
+
+```text
+# for rails
+
+FUSIONAUTH_LOCATION=http://localhost:9011
+
+CLIENT_ID=e9fdb985-9173-4e01-9d73-ac2d60d1dc8e
+```
+
+Utilizing the enviroment variables you just added we can now setup JSON Web Token (JWT) based authentication. Create a new file  `config/initializers/jwt_rack.rb`. This initializer is used to hold configuration settings that are made after all of the frameworks and plugins are loaded. 
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/config/initializers/jwt_rack.rb"}
+  lang="ruby"/>
+
+Having this code protects your endpoints from anonymous users and passes the JWT payload to the controller. The JWT payload includes `roles` encoded in the JWT you receive from FusionAuth. The decoded payload of a JWT for a `teller` might look like this:
+
+```json
+{
+  "aud": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "exp": 1689289585,
+  "iat": 1689285985,
+  "iss": "http://localhost:9011",
+  "sub": "00000000-0000-0000-0000-111111111111",
+  "jti": "ebaa4184-2320-47dd-925b-2e18756c635f",
+  "authenticationType": "PASSWORD",
+  "email": "teller@example.com",
+  "email_verified": true,
+  "applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "roles": [
+    "teller"
+  ],
+  "auth_time": 1689285985,
+  "tid": "d7d09513-a3f5-401c-9685-34ab6c552453"
+}
+```
+
+### Add Controllers
+
+Create a new file for the MakeChangeController `app/controllers/make_change_controller.rb`. This controller will verify that the authenticated user has either the `teller` or `customer` role. It then takes in the url parameter `total` to calculate which coins will be returned in the JSON payload.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/app/controllers/make_change_controller.rb"}
+  lang="ruby"/>
+
+Create a new file for the PanicController `app/controllers/panic_controller.rb`. This controller will verify that the authenticated user has only the `teller` role. It will respond with `Proper role not found for user.` if this role is not found. This route also verifies that this is a `POST` request and if this is not true it will respond with `Only POST method is supported.` If both of these tests are passed the controller will return a successful message of `We've called the police!`.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/app/controllers/panic_controller.rb"}
+  lang="ruby"/>
+
+## Run the API
+
+Start the API resource server by running:
+
+In a command shell window, navigate to the root directory of your application and run the following command to start the server.
+
+```shell
+bundle e rails s -p 4001
+```
+
+### Get a Token
+
+There are [several ways to acquire a token](/docs/v1/tech/oauth/) in FusionAuth, but for this example you will use the [Login API](/docs/v1/tech/apis/login) to keep things simple.
```

Comment:
Should mention here that you should open a new terminal window and will be using it window for the commands the following commands. The server is running in the terminal window I was using.